app/src/main/java/protect/card_locker/ScanActivity.kt

```diff
@@ -0,0 +1,549 @@
+            }
+
+        customBarcodeScannerBinding.fabOtherOptions.setOnClickListener {
+            setScannerActive(false)
+
+            val list: ArrayList<HashMap<String, Any>> = arrayListOf()
+            val texts = arrayOf(
+                getString(R.string.addWithoutBarcode),
+                getString(R.string.addManually),
+                getString(R.string.addFromImage),
+                getString(R.string.addFromPdfFile),
+                getString(R.string.addFromPkpass)
+            )
+            val icons = arrayOf(
+                R.drawable.baseline_block_24,
+                R.drawable.ic_edit,
+                R.drawable.baseline_image_24,
+                R.drawable.baseline_picture_as_pdf_24,
+                R.drawable.local_activity_24px
+            )
+            val columns = arrayOf("text", "icon")
+
+            for (i in 0 until texts.size) {
+                val map: HashMap<String, Any> = hashMapOf()
+                map.put(columns[0], texts[i])
+                map.put(columns[1], icons[i])
+                list.add(map)
+            }
+
+            val adapter: ListAdapter = SimpleAdapter(
+                this,
+                list,
+                R.layout.alertdialog_row_with_icon,
+                columns,
+                intArrayOf(R.id.textView, R.id.imageView)
+            )
+
+            val builder = MaterialAlertDialogBuilder(this)
+            builder.setTitle(getString(R.string.add_a_card_in_a_different_way))
+            builder.setAdapter(adapter) { _, i ->
+                when (i) {
+                    0 -> addWithoutBarcode()
+                    1 -> addManually()
+                    2 -> addFromImage()
+                    3 -> addFromPdf()
+                    4 -> addFromPkPass()
+                    else -> throw IllegalArgumentException(
+                        "Unknown 'Add a card in a different way' dialog option: $i"
+                    )
+                }
+            }
+            builder.setOnCancelListener { _ -> setScannerActive(true) }
+            builder.show()
+        }
+
+        // Configure barcodeScanner
+        barcodeScannerView = binding.zxingBarcodeScanner
+        val barcodeScannerIntent = Intent().apply {
+            putExtra(DecodeHintType.ALSO_INVERTED.name, java.lang.Boolean.TRUE)
+        }
+        barcodeScannerView.initializeFromIntent(barcodeScannerIntent)
+
+        // Even though we do the actual decoding with the barcodeScannerView
+        // CaptureManager needs to be running to show the camera and scanning bar
+        capture = CatimaCaptureManager(this, barcodeScannerView, this::onCaptureManagerError)
+        val captureIntent = Intent().apply {
+            putExtra(Intents.Scan.BEEP_ENABLED, false)
+        }
+        capture.initializeFromIntent(captureIntent, savedInstanceState)
+
+        barcodeScannerView.decodeSingle(object: BarcodeCallback {
+            override fun barcodeResult(result: BarcodeResult?) {
```

Comment:
Looks like result can't be null. That would be good if you fix (remove) all `!!` in file